TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/fragments/ComposeFragment.java

```diff
@@ -0,0 +1,198 @@
+            }
+        });
+    }
+
+    private void saveReview(Restaurant selectedRestaurant, ParseUser author, String text, float rating, boolean shareWithEveryone) {
+        ParseReview review = new ParseReview();
+        review.setAuthor(author);
+        review.setRating(rating);
+        review.setText(text);
+        review.setGlobal(shareWithEveryone);
+        checkRestaurantExistAndSave(selectedRestaurant, review);
+    }
+
+    private void checkRestaurantExistAndSave(final Restaurant selectedRestaurant, final ParseReview review) {
+        ParseQuery<ParseRestaurant> parseRestaurantQuery = ParseQuery.getQuery(ParseRestaurant.class);
+        parseRestaurantQuery.whereEqualTo(ParseRestaurant.ZOMATO_ID_KEY, selectedRestaurant.getId());
+        parseRestaurantQuery.findInBackground(new FindCallback<ParseRestaurant>() {
+            @Override
+            public void done(List<ParseRestaurant> restaurants, ParseException e) {
+                //if restaurant is not exist yet on Parse go ahead an save to Parse
+                //else get the exist restaurant and set the review points to that
+                if (restaurants.isEmpty() || e != null) {
+                    review.setRestaurant(SaveRestaurant(selectedRestaurant));
+                } else {
+                    review.setRestaurant(restaurants.get(0));
+                }
+                review.saveInBackground(new SaveCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e != null) {
+                            Log.e(TAG, "Error while saving review", e);
+                        } else {
+                            Log.i(TAG, "Review save was success!!");
+                        }
+                    }
+                });
+                return;
+            }
+        });
+    }
+
+    //Save restaurant to Parse
+    private ParseRestaurant SaveRestaurant(Restaurant selectedRestaurant) {
+        ParseRestaurant parseRestaurant = new ParseRestaurant(selectedRestaurant);
+        parseRestaurant.set();
+        parseRestaurant.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "Error while saving restaurant", e);
+                } else {
+                    Log.i(TAG, "Restaurant save was success!!");
+                }
+            }
+        });
+        return parseRestaurant;
+    }
+
+
+    @Override
+    public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
+        if(position == EVERYONE){
+            shareWithEveryone = true;
+        }else {
+            shareWithEveryone = false;
+        }
+        Log.i(TAG, position+" "+adapterView.getItemAtPosition(position));
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> adapterView) {
+
```

Comment:
Whenever you have an empty method, like this, it's usually a good idea to put some sort of comment inside to say why we're leaving it empty.